src/hooks/useLogin.ts

```diff
@@ -0,0 +1,20 @@
+import { useMutation } from "@tanstack/react-query";
+import { loginService } from "../services/authService";
+import { notifySuccess, notifyError } from "../utils/toastify";
+import { LoginFormType } from "../interfaces/authTypes";
+
+export const useLogin = (options: { onSuccess: () => void }) => {
+  return useMutation({
+    mutationFn: async (data: LoginFormType) => {
+      const userData = await loginService(data.username, data.password);
```

Comment:
You can send data object directly instead of destructuring
For example loginServices(data)